main.tf

```diff
@@ -290,21 +366,45 @@ resource "azurerm_monitor_diagnostic_setting" "pe_kv_nic" {
   depends_on                     = [azurerm_private_endpoint.pep]
   count                          = var.enabled && var.diagnostic_setting_enable && var.enable_private_endpoint ? 1 : 0
   name                           = format("%s-pe-kv-nic-diagnostic-log", module.labels.id)
-  target_resource_id             = element(azurerm_private_endpoint.pep[count.index].network_interface.*.id, count.index)
+  target_resource_id             = azurerm_private_endpoint.pep[count.index].network_interface[0].id
   storage_account_id             = var.storage_account_id
   eventhub_name                  = var.eventhub_name
   eventhub_authorization_rule_id = var.eventhub_authorization_rule_id
   log_analytics_workspace_id     = var.log_analytics_workspace_id
   log_analytics_destination_type = var.log_analytics_destination_type
-  metric {
-    category = "AllMetrics"
-    enabled  = var.Metric_enable
-    retention_policy {
-      enabled = var.retention_policy_enabled
-      days    = var.days
+  dynamic "metric" {
```

Comment:
Don't we require `enabled_log` for this?

Reply:
No . its not support for network interface card diagnosis setting.